libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -273,6 +287,25 @@ void AP_AIS::send(mavlink_channel_t chan)
     }
 }
 
+void AP_AIS::handle_message(const mavlink_ais_vessel_t &packet)
+{
+    if (!enabled()) {
+        return;
+    }
+
+    uint16_t index;
+    if (!get_vessel_index(packet.MMSI, index)) {
+        // not found and not enough memory to add it
+        return;
+    }
+    memcpy(&_list[index].info, &packet, sizeof(mavlink_ais_vessel_t));
+    _list[index].last_update_ms = AP_HAL::millis();
```

Comment:
It might also be a good idea to include the time since last communication value.
```suggestion
    _list[index].last_update_ms = AP_HAL::millis() - packet.tslc * 1000;
```

Maybe we should also check that the time is more recent than the current data for that vessel. I'm just thinking of a case where we have a local receiver and are getting MAVLink packets for the same vessel.